packages/core/contracts-ovm/insured-bridge/external/AVM_CrossDomainEnabled.sol

```diff
@@ -0,0 +1,59 @@
+pragma solidity ^0.8.0;
+
+import "./iAVM_Inbox.sol";
+
+abstract contract AVM_CrossDomainEnabled {
+    modifier onlyFromCrossDomainAccount(address l1Counterpart) {
+        require(msg.sender == applyL1ToL2Alias(l1Counterpart), "ONLY_COUNTERPART_GATEWAY");
+        _;
+    }
+
+    uint160 constant offset = uint160(0x1111000000000000000000000000000000001111);
+    iAVM_Inbox public immutable inbox;
+
+    /**
+     * @param _inbox Contract that sends generalized messages to the Arbitrum chain.
+     */
+    constructor(address _inbox) {
+        inbox = iAVM_Inbox(_inbox);
+    }
+
+    // l1 addresses are transformed during l1->l2 calls. see https://developer.offchainlabs.com/docs/l1_l2_messages#address-aliasing for more information.
+    function applyL1ToL2Alias(address l1Address) internal pure returns (address l2Address) {
```

Comment:
same as for this function

Reply:
actually why are you not re-using this file? https://github.com/UMAprotocol/protocol/blob/7d73fde9cf062c50a599166a0fe52f3f2a817ddb/packages/core/contracts-ovm/insured-bridge/external/arbitrum/AVM_CrossDomainEnabled.sol#L3

you simply need to add the one additional function. much is the same, no?

Reply:
See above response